### PR TITLE
Made it simpler to use your own GPX file on startup

### LINKED ISSUE #105: Make it simpler to use your own GPX file on startup
It needs to be simpler to use your own GPX file:
- For the user who wants to use an arbitrary GPX
- For the IT professional who wants to point the app to a GPX without writing code
- For the developer who wants to build their own GPX into the app

However, the precedence should be for real users using hardware location capabilities.

Startup rules:
- If the user has previously selected a location mode--hardware, built-in simulation, or selected GPX file that still exists--use the most recently selected location mode.
- If the most recently selected location mode is a GPX file that no longer exists, change the mode to built-in simulation.
- If no location mode has been selected:
  - If the device has a file called simulation.gpx in the SquadLeader directory, use it in simulation mode.
  - If SquadLeader/simulation.gpx does not exist, use hardware mode.
  - In either case, don't set the location mode as a user preference.

Resetting the map should *not* reset the location mode.

### PR DESCRIPTION
**Pull request by garys-esri**
Fixed #105

@csmoore, please review and merge at your convenience.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,10 @@ To build the app from source:
 * Uses ArcGIS Runtime 10.2.7
 * Requires Android 4.0.3 or higher
 * Added Mil2525CMessageLayer layer type for GeoMessage files (see the [User Guide](documentation/UserGuide.md#geomessage-files-and-layers))
+* Refactored the location capabilities
+  * TODO users
```
Comment (csmoore) [inline README.md]: TODOs are fine even in Readme's (I actually wish people would use them more, it is much preferable to potentially forgetting)

Comment (garys-esri) [inline README.md]: @csmoore oops. This one was an oversight. Let me address before merge.

**Comment by csmoore** [comment]
Just visually reviewed but look good to merge (I'll wait for you to change/commit the TODO to a TO-DONE)

**Comment by garys-esri** [comment]
@csmoore, I'm sorry but I created this pull request prematurely. I need to account for the case where a deployer puts a simulation.gpx file on the device to override the user's preference or defaults. I'll let you know when it's ready. I apologize.

**Comment by garys-esri** [comment]
@csmoore, this is now ready for review. Thanks for your patience.

File: documentation/UserGuide.md [hunk 2]
```diff
@@ -65,6 +65,12 @@ If you do not provide a mapconfig.xml file, a default list of ArcGIS Online base
...
 
+#### GPS simulation
+
+By default, Squad Leader uses the device's location capabilities to display the user's current location on the map. This is the appropriate setting for real-world deployments, so most likely you don't need to do anything to change location settings. Read on if you are interesting in using simulated GPS.
+
+Squad Leader can use a user-specified GPX file or a built-in GPX file of track points in Monterey, California. An IT professional who wants to deploy Squad Leader with a simulation based on a GPX file can create a file named simulation.gpx, create a directory called SquadLeader on the device's main storage (not an SD card), and put simulation.gpx in the SquadLeader directory. Then Squad Leader will use simulation.gpx to simulate the user's location until the user chooses a location setting in the Squad Leader UI. After a user has chosen a location setting, simulation.gpx will not be used again unless the Squad Leader app's data is cleared in Android settings.
```
Comment (csmoore) [inline documentation/UserGuide.md]: Just a future maintenance tip, if you don't include/mention specifics like the "Monterey, California" area of interest, this doc is still complete andl reads the same, but doesn't need to be updated if this changes (& it probably will change at some point again)

**Comment by csmoore** [comment]
Looks good

File: documentation/UserGuide.md (hunk 2 above)
Comment (garys-esri) [inline documentation/UserGuide.md]: Good idea. I created #135 to capture this maintenance need.

**Pull request merged**